sql-plugin/src/main/java/com/nvidia/spark/rapids/InternalRowToColumnarBatchIterator.java

```diff
@@ -202,6 +207,50 @@ public ColumnarBatch next() {
     }
   }
 
+  private Tuple2<SpillableHostBuffer[], AutoCloseableTargetSize>
+  allocBuffers(HostMemoryBuffer[] hBufs, SpillableHostBuffer[] sBufs,
+      AutoCloseableTargetSize numRowsWrapper) {
+    long dataBytes = calcDataLengthEstimate((int) numRowsWrapper.targetSize());
+    long offsetBytes = calcOffsetLengthEstimate((int) numRowsWrapper.targetSize());
+    hBufs[0] = HostAlloc$.MODULE$.alloc(dataBytes, true);
+    sBufs[0] = SpillableHostBuffer$.MODULE$.apply(hBufs[0], hBufs[0].getLength(),
+        SpillPriorities$.MODULE$.ACTIVE_ON_DECK_PRIORITY(),
+        RapidsBufferCatalog$.MODULE$.singleton());
+    hBufs[0] = null; // Was closed by spillable
+    hBufs[1] = HostAlloc$.MODULE$.alloc(offsetBytes, true);
```

Comment:
do we need to wrap this in a try/catch so we can close sBufs[0] if `alloc` throws?

Reply:
oh nm, I think `allocBuffersWithRestore` takes care of it.

Reply:
I think there may be a problem here actually.  If we throw with something other than a retry exception, the withRestoreOnRetry won't handle it.